### ISSUE TITLE
Error middlewares don't run if layer has parent

### ISSUE THREAD

**Issue by masterada** [h2non/gentleman#35]
Steps to reproduce:

1. create a client
2. create a request from the client
3. register middleware for error phase for the request (not the client!)
4. trigger an error (eg: call a random url: localhost:12345)

Registered middleware won't run.

In middleware.go, Layer.Run method:

```go
	ctx = s.parent.Run(phase, ctx)
	if ctx.Error != nil || ctx.Stopped {
		return ctx
	}
```

This will cause middlewares registered for error phase not to run if there is an error.

Possible solution:
```go
	ctx = s.parent.Run(phase, ctx)
	if (phase != "error" && ctx.Error != nil) || ctx.Stopped {
		return ctx
	}
```
but I'm not sure what that would break.

**Comment by tomas-fp** [comment]
yeah, this sounds likely to be a execution bug. In such case, we would need to run the middleware phase even if the flow was stopped. I will review it and come up with a fix soon.

**Comment by h2non** [comment]
Fixed in `gentleman@v2.0.1`. Try upgrading the package:
```bash
go get -u gopkg.in/h2non/gentleman.v2
```

You can see a working example here:
https://github.com/h2non/gentleman/blob/master/_examples/error-handler/error.go

**Comment by masterada** [comment]
Thanks
